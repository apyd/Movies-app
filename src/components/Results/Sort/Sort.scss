@use "../../../styles/colors";

.sort {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  font-size: inherit;
  text-transform: uppercase;
  gap: 0 1rem;
  position: relative;
  font-size: 0.9rem;

  &::after {
    content: "\25BC";
    position: absolute;
    right: -1rem;
    color: colors.$pastel-red;
  }

  &__label {
    color: colors.$alt-grey;
    letter-spacing: 0.1ch;
    white-space: nowrap;
  }

  &__input {
    background: transparent;
    border: none;
    -webkit-appearance: none;
    appearance: none;
    color: colors.$white;
    font-size: inherit;
    font-family: inherit;
    font-weight: 700;
    letter-spacing: 0.1ch;
    text-transform: uppercase;
  }
}
