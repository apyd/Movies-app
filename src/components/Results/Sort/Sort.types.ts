export interface ISortProps {
    options: Array<{
        id: number,
        label: string,
        value: string | number,
        isDefault?: boolean
    }>,
    onOptionChange: () => void
    sortLabel: string
}