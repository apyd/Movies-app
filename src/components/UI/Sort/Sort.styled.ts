import styled from '@emotion/styled';
import { colors } from '../../../shared/colors.styled';

export const SortWrapper = styled.div`
  display: flex;
  align-items: center;
  gap: 0 1rem;
  position: relative;
  text-transform: uppercase;
  font-size: 0.9rem;

  &::after {
    position: absolute;
    right: -1rem;
    content: '\\25BC';
    color: ${colors.pastelRed};
  }
`;

export const SortLabel = styled.label`
  color: ${colors.altGrey};
  letter-spacing: 0.1ch;
  white-space: nowrap;
`;

export const SortSelect = styled.select`
  appearance: none;
  border: none;
  background: transparent;
  color: ${colors.white};
  font-size: inherit;
  font-family: inherit;
  font-weight: 700;
  text-transform: uppercase;
  letter-spacing: 0.1ch;
`;
