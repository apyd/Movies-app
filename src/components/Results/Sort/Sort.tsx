import React, { FC } from "react";
import classNames from "classnames/bind";
import { ISortProps } from "./Sort.types";
import styles from "./Sort.scss";

export const Sort: FC<ISortProps> = ({
  options,
  onOptionChange,
  sortLabel = "Sort by",
}) => {
  const cx = classNames.bind(styles);
  return (
    <div className={cx("sort")}>
      <label className={cx("sort__label")} htmlFor="sort">
        {sortLabel}
      </label>
      <select
        className={cx("sort__input")}
        name="sort"
        id="sort"
        onChange={(e) => onOptionChange(e.target.value)}
      >
        {options &&
          options.map(({ id, label, value }) => (
            <option key={id} value={value}>
              {label}
            </option>
          ))}
      </select>
    </div>
  );
};
