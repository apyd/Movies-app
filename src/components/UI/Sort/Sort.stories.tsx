import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Sort } from './Sort';
import { SORT_OPTIONS } from '../../../dictionary/dictionary';

export default {
  title: 'components/UI/Sort',
  component: Sort,
  argTypes: {
    options: {
      options: ['average_vote', 'release_date'],
      control: { type: 'select' }
    }
  }
} as ComponentMeta<typeof Sort>;

const Template: ComponentStory<typeof Sort> = (args) => <Sort {...args} />;

export const Default = Template.bind({});
Default.args = {
  selected: SORT_OPTIONS[0].value,
  onOptionChange: () => {},
  sortLabel: 'Sort by'
};
