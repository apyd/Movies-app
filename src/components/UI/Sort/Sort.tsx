import React, { FC } from 'react';
import { SortLabel, SortSelect, SortWrapper } from './Sort.styled';
import { ISortProps } from './Sort.types';

export const Sort: FC<ISortProps> = ({
  selected,
  options,
  onOptionChange,
  sortLabel = 'Sort by'
}) => {
  return (
    <SortWrapper data-testid="Sort">
      <SortLabel htmlFor="sort">{sortLabel}</SortLabel>
      <SortSelect
        data-testid="SortSelect"
        name="sort"
        id="sort"
        value={selected}
        onChange={(e) => onOptionChange(e.target.value)}>
        {options &&
          options.map(({ id, label, value }) => (
            <option data-testid={id} key={id} value={value}>
              {label}
            </option>
          ))}
      </SortSelect>
    </SortWrapper>
  );
};
