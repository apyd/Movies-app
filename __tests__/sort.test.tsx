import { expect } from '@jest/globals';
import { render } from '@testing-library/react';
import { Sort } from '../src/components/UI/Sort/Sort';
import { SORT_OPTIONS } from '../src/dictionary/dictionary';
import { ISortProps } from '../src/components/UI/Sort/Sort.types';

const renderer = require('react-test-renderer');

describe('<Sort/>', () => {
  let props: ISortProps;

  beforeEach(() => {
    props = {
      options: SORT_OPTIONS,
      selected: SORT_OPTIONS[0].value,
      onOptionChange: jest.fn(),
      sortLabel: 'Sort by'
    };
  });

  it('snapshot testing suite matches DOM Snapshot', () => {
    const domTree = renderer.create(<Sort {...props} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    expect.hasAssertions();
    const { getByTestId } = render(<Sort {...props} />);
    const sort = getByTestId('Sort');
    //@ts-ignore
    expect(sort).toBeInTheDocument();
  });

  it('renders when empty selected prop is passed', () => {
    expect.hasAssertions();

    props = {
      ...props,
      selected: ''
    };

    const { getByTestId } = render(<Sort {...props} />);
    const sort = getByTestId('Sort');
    //@ts-ignore
    expect(sort).toBeInTheDocument();
  });
});
