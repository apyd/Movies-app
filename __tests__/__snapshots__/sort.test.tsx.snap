// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort/> snapshot testing suite matches DOM Snapshot 1`] = `
<div
  className="css-1jmft5i"
  data-testid="Sort"
>
  <label
    className="css-xf46ac"
    htmlFor="sort"
  >
    Sort by
  </label>
  <select
    className="css-17fdvkm"
    data-testid="SortSelect"
    id="sort"
    name="sort"
    onChange={[Function]}
    value="average_vote"
  >
    <option
      data-testid={1}
      value="average_vote"
    >
      Rating
    </option>
    <option
      data-testid={2}
      value="release_date"
    >
      Release date
    </option>
  </select>
</div>
`;
